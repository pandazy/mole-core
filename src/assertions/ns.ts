import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
