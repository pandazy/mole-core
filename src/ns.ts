export function ns(context: string): string {
  return `Mole:${context}`;
}
