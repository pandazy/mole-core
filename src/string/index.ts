export * from './replace-all';
export * from './search-all';
export * from './stringify';
