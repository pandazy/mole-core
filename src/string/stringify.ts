export const stringify = JSON.stringify.bind(JSON);
