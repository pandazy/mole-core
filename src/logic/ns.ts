import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
