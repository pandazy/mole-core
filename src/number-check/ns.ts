import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
