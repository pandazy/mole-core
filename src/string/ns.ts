import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
