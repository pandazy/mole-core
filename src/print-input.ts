import { isA, Nil } from './type-check/is-a';
import { stringify } from './string/stringify';
import { cond, when, otherwise } from './logic';

const printFunc = (ag: any) => `<Function: ${ag.name || 'anonymous'}>`;
const printDesc = (ag: any) => ag.toString();
const printNull = (ag: any) => `${ag}`;
const printJSON = (ag: any) => stringify(ag);

export const printInput: (input: any) => string = cond<string>([
	when(isA(String))(stringify),
	when(isA(Function))(printFunc),
	when(isA(Object))(printJSON),
	when(isA(Nil))(printNull),
	otherwise(printDesc),
]);
