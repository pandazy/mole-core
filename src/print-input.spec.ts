import { printInput } from './print-input';
import { ns } from './ns';
import { stringify } from './string/stringify';

describe(ns('print-input'), () => {
	it('should print an arbitrary value', () => {
		expect.assertions(6);

		expect(printInput(() => 'foo')).toMatchSnapshot();
		expect(
			printInput(function callIt() {
				return 'bar';
			}),
		).toMatchSnapshot();
		expect(printInput({ name: 'John' })).toBe(stringify({ name: 'John' }));
		expect(printInput('foo')).toBe(stringify('foo'));
		expect(printInput(null)).toBe('null');
		expect(printInput(false)).toBe('false');
	});
});
