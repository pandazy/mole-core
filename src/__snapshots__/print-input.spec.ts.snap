// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:print-input should print an arbitrary value 1`] = `"<Function: anonymous>"`;

exports[`Mole:print-input should print an arbitrary value 2`] = `"<Function: callIt>"`;
